bundles/org.openhab.binding.mielecloud/src/main/java/org/openhab/binding/mielecloud/internal/handler/MieleBridgeHandler.java

```diff
@@ -0,0 +1,359 @@
+        scheduler.schedule(() -> {
+            try {
+                getWebservice().logout();
+            } catch (Exception exception) {
+                logger.warn("Failed to logout from Miele cloud.");
+                logger.debug("Exception details:", exception);
+            }
+            OptionalUtils.ofNullable(logoutFuture).map(future -> future.complete(null));
+        }, 1, TimeUnit.NANOSECONDS);
+    }
+
+    private void tryInitializeWebservice() {
+        Optional<String> accessToken = tokenRefresher.getAccessTokenFromStorage(getOAuthServiceHandle());
+        if (!accessToken.isPresent()) {
+            logger.info("No OAuth2 access token available. Retrying later.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    I18NKeys.BRIDGE_STATUS_DESCRIPTION_ACCESS_TOKEN_NOT_CONFIGURED);
+            return;
+        }
+        getWebservice().setAccessToken(accessToken.get());
+        updateProperty(MieleCloudBindingConstants.PROPERTY_ACCESS_TOKEN, accessToken.get());
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        synchronized (this) {
+            if (getThing().getStatus() == ThingStatus.REMOVING && status != ThingStatus.REMOVED) {
+                return;
+            }
+
+            super.updateStatus(status, statusDetail, description);
+        }
+    }
+
+    @Override
+    public void onConnectionAlive() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void onConnectionError(ConnectionError connectionError, int failedReconnectionAttempts) {
+        if (connectionError == ConnectionError.AUTHORIZATION_FAILED) {
+            tryToRefreshAccessToken();
+            return;
+        }
+
+        if (failedReconnectionAttempts <= NUMBER_OF_SSE_RECONNECTION_ATTEMPTS_BEFORE_STATUS_IS_UPDATED
+                && getThing().getStatus() != ThingStatus.UNKNOWN) {
+            return;
+        }
+
+        if (getThing().getStatus() == ThingStatus.UNKNOWN && connectionError == ConnectionError.REQUEST_INTERRUPTED
+                && failedReconnectionAttempts <= NUMBER_OF_SSE_RECONNECTION_ATTEMPTS_BEFORE_STATUS_IS_UPDATED) {
+            return;
+        }
+
+        switch (connectionError) {
+            case AUTHORIZATION_FAILED:
+                // Handled above.
+                break;
+
+            case REQUEST_EXECUTION_FAILED:
+            case SERVICE_UNAVAILABLE:
+            case RESPONSE_MALFORMED:
+            case TIMEOUT:
+            case TOO_MANY_RERQUESTS:
+            case SSE_STREAM_ENDED:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                break;
+
+            case SERVER_ERROR:
+            case REQUEST_INTERRUPTED:
+            case OTHER_HTTP_ERROR:
+            default:
+                updateStatus(ThingStatus.OFFLINE);
```

Comment:
I think it wouldn't be helpful as this case handles
- HTTP errors we cannot handle any other way than retrying later (what we do)
- Interruptions while sending a request (likely because openHAB is shutting down)

We could include some generic messages like "Request to cloud service failed", but does that really provide reasonable assistance to the user?

Reply:
You could set a generic message that this is a transient error. InterruptedExceptions might not be seen by the user at all in the UI.

Reply:
I added an error message. Please check again.